src/function/scalar/generic/current_setting.cpp

```diff
@@ -39,13 +39,19 @@ unique_ptr<FunctionData> CurrentSettingBind(ClientContext &context, ScalarFuncti
 		throw ParserException("Key name for struct_extract needs to be neither NULL nor empty");
 	}
 
-	auto &config_map = context.db->config.set_variables;
-	if (config_map.find(key_val.str_value) == config_map.end()) {
-		throw InvalidInputException("Variable '%s' was not SET in this context", key_val.str_value);
+	const auto &key = key_val.str_value;
+	auto &session_config_map = context.set_variables;
+	auto &global_config_map = context.db->config.set_variables;
+
+	bool has_session_value = session_config_map.find(key) != session_config_map.end();
+	bool has_global_value = global_config_map.find(key) != global_config_map.end();
+	if (has_session_value || has_global_value) {
+		Value val = has_session_value ? session_config_map[key] : global_config_map[key];
```

Comment:
This performs a double read of the unordered_map and will unnecessarily probe the global config map if there is a session value present. Something like this will be more efficient:

```cpp
auto session_entry = session_config_map.find(key);
if (session_entry != session_config_map.end()) {
   return session_entry->second;
}
auto global_entry = global_config_map.find(key);
if (global_entry != global_config_map.find.end()) {
  return global_entry->second;
}
throw InvalidInputException("...");
```

Reply:
Done.